docs/RELEASE_NOTES.md

```diff
@@ -146,287 +146,277 @@ Use Cocos Console to create a new game:
+    Added a method to the Application class to get the version of the app on iOS/Mac/Android/Win8, on the remaining platforms that don't have app versions it should return an empty string.
 
-2. ScrollView performance on Web engine
+    The feature has been requested in this discussion thread: http://discuss.cocos2d-x.org/t/check-app-version/6294/5
 
-    ScrollView and ListView are the popular UI controls in Web engine. Their
-    performance is not perfect in previous versions, especially when there are multiple sub-controls. In v3.9, we have improved its rendering performance. They only act on the contents displayed on the current screen. Test
-    date shows that, comparing with v3.8, rendering efficiency of v3.9 have been improved for twice to four times in different devices and browsers. 
+2. Set focus to Widget when touched.
 
-3. Scale9Sprite performance on Web engine
+    Automatically set focus for widgets if focus enabled.
 
-    In this version, we have changed the way to construct 9-slice sprite. The engine uses 9 rendering commands instead of the 9 nodes in previous versions. This helps to reduce memory usage and improve rendering performance. 
 
-4. Decoupled Sprite's setTexture and updateColor in Web engine.
+3. Improved JS bindings with more secured memory management.
 
-    - Organized the rendering logic in Sprite. UpdateColor is accomplished by texture instead of the Sprite. 
-    - Fixed a bug about image with alpha channel that when the image is set to black, there is color difference between previous and current version. 
-    - Improved texture update logic to reduce texture updates when changing colors.
-    - Improved the logic about the rendering function in SpriteCanvasRenderCmd.
-    - Removed some duplicate codes about updateColor.
+    Improvement covered both auto bindings and manual bindings.
+    All changes follow Generational GC style.
 
-5. Support for debugging and release on real devices with Xcode7 and iOS9
+    The background of this feature: [http://blogs.msdn.com/b/abhinaba/archive/2009/03/02/back-to-basics-generational-garbage-collection.aspx](http://blogs.msdn.com/b/abhinaba/archive/2009/03/02/back-to-basics-generational-garbage-collection.aspx "http://blogs.msdn.com/b/abhinaba/archive/2009/03/02/back-to-basics-generational-garbage-collection.aspx")
 
-    In v3.8.1, we have made it possible to debug on Xcode7. However, there was a bug with iOS9 real device debuging, and in v3.9, we have fixed the bug.
+    This requires to adopt the new GC coding style in all JSB code.
 
+    The new GC coding style can be verified with this tool: [https://github.com/joshuastray/RootAnalysis](https://github.com/joshuastray/RootAnalysis "https://github.com/joshuastray/RootAnalysis")
 
 ## Other changes
+[NEW]           UIText::clone supports clone the text effect.
```

Comment:
And I think it's no need to duplicated the content which are already in **CHANGLOG**.
In release notes, I think we should provide more details of each new feature.

Some code snippets and screenshot would be great.

The `ui::Scale9Sprite` improvement and three types of overflow in `cc:Label` should be detailed explained.